org-mode/lib/Data/Org.hs

```diff
@@ -125,7 +142,64 @@ data Block
   | Table (NonEmpty Row)
   | Paragraph (NonEmpty Words)
   deriving stock (Eq, Show, Generic)
-  deriving anyclass (Hashable)
+
+-- | An org-mode timestamp. Must contain at least a year-month-day and the day
+-- of the week:
+--
+-- @
+-- \<2021-04-27 Tue\>
+-- @
+--
+-- but also may contain a time:
+--
+-- @
+-- \<2021-04-27 Tue 12:00\>
+-- @
+--
+-- or a time range:
+--
+-- @
+-- \<2021-04-27 Tue 12:00-13:00\>
+-- @
+--
+-- and/or a repeater value:
+--
+-- @
+-- \<2021-04-27 Tue +1w\>
+-- @
+data OrgDateTime = OrgDateTime
+  { dateDay       :: Day
+  , dateDayOfWeek :: DayOfWeek
+  , dateTime      :: Maybe OrgTime
+  , dateRepeat    :: Maybe Repeater }
+  deriving stock (Eq, Show)
```

Comment:
`Ord` too? For this and other types.

Reply:
I suppose `OrgTime` can be compared on `timeStart`, and `OrgDateTime` on `(dateDay, dateTime, dateRepeat)`.